stl/inc/algorithm

```diff
@@ -3436,6 +3436,14 @@ _CONSTEXPR20 _FwdIt2 swap_ranges(const _FwdIt1 _First1, const _FwdIt1 _Last1, _F
     const auto _UFirst1 = _Get_unwrapped(_First1);
     const auto _ULast1  = _Get_unwrapped(_Last1);
     const auto _UFirst2 = _Get_unwrapped_n(_First2, _Idl_distance<_FwdIt1>(_UFirst1, _ULast1));
+
+#if _ITERATOR_DEBUG_LEVEL == 2
+    if constexpr (_Is_random_iter_v<_FwdIt1> && _Is_random_iter_v<_FwdIt2>) {

```

Comment:
I believe this needs to be guarded with `#if _HAS_IF_CONSTEXPR` (see other comment), as `swap_ranges` is available in C++14 mode. (C++17-and-above code can inherently assume that `if constexpr` is available.)

Reply:
ah, I had a quick look at [here](https://en.cppreference.com/w/cpp/algorithm/swap_ranges), and I thought it's a C++17 feature! Thank you.